accounts/src/store/accounts_db.rs

```diff
@@ -0,0 +1,829 @@
+        }
+    }
+
+    pub fn new_for_testing() -> Self {
+        let db = SolanaAccountsDb::new_single_for_tests();
+        let accounts = Accounts::new(Arc::new(db));
+        Self {
+            accounts,
+            processed_slot: AtomicU64::new(0),
+            confirmed_slot: AtomicU64::new(0),
+            finalised_slot: AtomicU64::new(0),
+        }
+    }
+}
+
+#[async_trait]
+impl AccountStorageInterface for AccountsDb {
+    async fn update_account(&self, account_data: AccountData, _commitment: Commitment) -> bool {
+        self.initilize_or_update_account(account_data).await;
+        true
+    }
+
+    async fn initilize_or_update_account(&self, account_data: AccountData) {
+        let shared_data = account_data.account.to_account_shared_data();
+        let account_to_store = [(&account_data.pubkey, &shared_data)];
+        self.accounts
+            .store_accounts_cached((account_data.updated_slot, account_to_store.as_slice()));
+    }
+
+    async fn get_account(
+        &self,
+        account_pk: Pubkey,
+        commitment: Commitment,
+    ) -> Result<Option<AccountData>, AccountLoadingError> {
+        let ancestors = self.get_ancestors_from_commitment(commitment);
+        Ok(self
+            .accounts
+            .load_with_fixed_root(&ancestors, &account_pk)
+            .map(|(shared_data, slot)| {
+                Self::convert_to_account_data(account_pk, slot, shared_data)
+            }))
+    }
+
+    async fn get_program_accounts(
+        &self,
+        program_pubkey: Pubkey,
+        account_filter: Option<Vec<RpcFilterType>>,
+        commitment: Commitment,
+    ) -> Option<Vec<AccountData>> {
+        let slot = self.get_slot_from_commitment(commitment);
+
+        let filter = |data: &AccountSharedData| {
+            match &account_filter {
+                Some(filters) => {
+                    filters.iter().all(|filter| {
+                        match filter {
+                            RpcFilterType::DataSize(size) => data.data().len() == *size as usize,
+                            RpcFilterType::Memcmp(cmp) => cmp.bytes_match(data.data()),
+                            RpcFilterType::TokenAccountState => unimplemented!(), // FIXME
+                        }
+                    })
+                }
+                None => true,
+            }
+        };
+
+        let transaction_accounts: Vec<TransactionAccount> = self
+            .accounts
+            .load_by_program_slot(slot, Some(&program_pubkey))
+            .into_iter()
+            .filter(|ta| filter(&ta.1))
+            .collect();
+
+        let result = transaction_accounts
+            .into_iter()
+            .map(|ta| Self::convert_to_account_data(ta.0, slot, ta.1))
+            .collect_vec();
+
+        if result.is_empty() {
+            None
+        } else {
+            Some(result)
+        }
+    }
+
+    async fn process_slot_data(&self, slot: Slot, commitment: Commitment) -> Vec<AccountData> {
```

Comment:
pratik promised to do a proper tested implementation in solana-lite-account-manager